Keas.Mvc/Controllers/Api/EquipmentController.cs

```diff
@@ -1,447 +1,452 @@
+                .Include(x => x.Team)
+                .SingleAsync(x => x.Id == id);
+
+            if (equipment == null)
+            {
+                return NotFound();
+            }
+
+            if (!equipment.Active)
+            {
+                return BadRequest(ModelState);
+            }
+
+            using (var transaction = _context.Database.BeginTransaction())
+            {
+
+                _context.Equipment.Update(equipment);
+
+                if (equipment.Assignment != null)
+                {
+                    await _eventService.TrackUnAssignEquipment(equipment); // call before we remove person info
+                    _context.EquipmentAssignments.Remove(equipment.Assignment);
+                }
+
+                equipment.Active = false;
+                await _eventService.TrackEquipmentDeleted(equipment);
+                await _context.SaveChangesAsync();
+                transaction.Commit();
+                return Json(null);
+            }
+
+        }
+
+        /// <summary>
+        /// Takes the top 5 history records
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="max">Defaults to 5</param>
+        /// <returns></returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(IEnumerable<History>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetHistory(int id, int max = 5)
+        {
+            var history = await _context.Histories
+                .Where(x => x.AssetType == "Equipment" && x.Equipment.Team.Slug == Team && x.EquipmentId == id)
+                .OrderByDescending(x => x.ActedDate)
+                .Take(max)
+                .AsNoTracking().ToListAsync();
+
+            return Json(history);
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(typeof(IEnumerable<ServiceNowPropertyWrapper>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetComputer(string id)
+        {
+            try
+            {
+                var result = await this._bigfixService.GetComputer(id);
+                var results = await this._serviceNowService.GetComputer(id);
+                
+                return Json(results);
+            }
+            catch (BigfixApiException ex)
+            {
+                if (ex.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    return NotFound();
+                }
+                else if (ex.StatusCode == System.Net.HttpStatusCode.Forbidden)
+                {
+                    return StatusCode(403);
+                }
+                else
+                {
+                    throw;
+                }
+            }
+        }
+
+        [HttpGet]
+        [ProducesResponseType(typeof(IEnumerable<BigfixComputerSearchResult>), StatusCodes.Status200OK)]
+        public async Task<IActionResult> GetComputersBySearch(string field, string value)

```

Comment:
this is still using bigfix so i'm guessing this is work in progress?

Reply:
Yea this one is still a work in progress